### ISSUE TITLE
Report an error for `d.pop('<required key>')` for `TypedDict`s

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#3562]
**Is your feature request related to a problem? Please describe.**

```py
from typing import TypedDict

class Foo(TypedDict):
    foo: int
    bar: str

def takes_foo(item: Foo) -> None:
    ...

foo: Foo = {"foo": 1, "bar": "bar"}
foo.pop("bar")  # should report an error
takes_foo(foo)
```

Removing a required key makes the `TypedDict` no longer valid.

**Additional context**

Mypy reports an error here:

```
error: Key "bar" of TypedDict "Foo" cannot be deleted
Found 1 error in 1 file (checked 1 source file)
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

The resulting error message is admittedly not as clear as the error message provided by mypy in this circumstance, but that's because of difference in the way the two type checkers implement the TypedDict feature.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.254, which I just published. It will also be included in the next release of pylance.
